text/0000-sigstore-integration.md

```diff
@@ -0,0 +1,387 @@
+
+When considering artifact provenance, one of the long known weaknesses in the OSS world are the central official package management systems that serve as a nexus for sharing and consumption of artifacts in any given language ecosystem. Historically these systems have struggled with a variety of attack vectors and suffered most specifically from a lack of concrete cryptographic artifact identities and security focused policies and tooling to manage contribution and ongoing artifact maintenance. The challenges are further exacerbated by the state of build system fidelity broadly. Provenance generation at artifact creation time and subsequent verification is where Sigstore comes in. Its approach, leveraging OIDC based identities as a gate to the signing function as opposed to long-term sensitive keys, as well as a public root and widely available verification capabilities make it a developer friendly, potentially ubiquitous signing architecture.
+
+This document will address two primary use case scenarios:
+
+1. The developer building Crates locally on their desktop
+1. Crates produced through a formal build plant (GitHub Actions, CircleCI, etc. and commercial enterprise internal build plants)
+
+These use cases will allow us to start to draw trust boundaries around artifact production processes and identify the metadata attributes that will need to be generated at artifact build-time which are subsequently consumed at run-time for effective policy and consumption controls to be possible for the ultimate consumers.
+
+It’s worth noting that the team at GitHub working on the NPM implementation have essentially landed at the same to use case patterns. In our discussions with them as this document was prepared they articulated a focus on the build plant pattern as that pattern allows for the establishment of an attestation story around the systems where the artifact was produced. Conversely, artifacts produced on a developer desktop or system that is not formally controlled from a security perspective must necessarily be classified as technically “unsafe”.
+
+In all of this the elephant in the room is identity. However, identity can also be addressed in the context of the two primary use case buckets. The easy one is build system identity, the hard one is individual contributor identity. This will be addressed in the [Identity section](#identities) below.
+
+This proposal has been influenced by [this document](https://docs.google.com/document/d/1mXrVAkUA9dd4M7fa_AJC8mQ55YnYJ-DKsGq30lh0FvA/edit#heading=h.jyrb6etgzah), as well as the work at GH regarding the NPM ecosystem, the RFC for which can be [found here](https://github.com/npm/rfcs/blob/main/accepted/0049-link-packages-to-source-and-build.md). Further we have engaged both parties in discussions to further glean relevant information related to this effort and to achieve a certain level of commonality in the overall pattern of implementation in the language ecosystem. 
+
+# Motivation
```

Comment:
This also follows from my comment on the summary: I think that the motivation for supply chain security is less interesting than the specific explanation of what SigStore is, and why it's chosen over the alternatives.

RFCs aren't tasks to be handed off by managers to developers. I further question the notion that tasks should settle on a specific implementation (e.g. SigStore) rather than being open to multiple (e.g. "implement supply-chain protections") and an RFC makes this especially relevant. In this RFC, you're fighting two fights: first, should we have supply-chain protections at all, and second, should we integrate SigStore. The first is likely to already be an uphill battle, and the second turns that hill into a wall.

Reply:
> I further question the notion that tasks should settle on a specific implementation (e.g. SigStore) rather than being open to multiple (e.g. "implement supply-chain protections") and an RFC makes this especially relevant.

I don't think "pick one" is an unreasonable position to take. Whether sigstore is the *right* one is a reasonable question, but I think Open Source far too often defaults to "pick anything you like as long as you're willing to glue it all together" rather than "here's one thing that's fully integrated and works well". We've already seen in this issue alone people mention that they don't want to deal with the details of signature requirements. We don't want people having to *select* those details.

I personally would advocate that we *do* pick *one* signing solution and make it work very well, rather than supporting several.